internal/store/root/move.go

```diff
@@ -156,6 +147,43 @@ func (r *Store) moveFromTo(ctx context.Context, subFrom *leaf.Store, from, to, f
 	return nil
 }
 
+func computeMoveDestination(src, from, to string, srcIsDir, dstIsDir bool) string {
+	// special case: moving up to the root
+	if to == "." || to == "/" {
+		dstIsDir = false
+		to = ""
+	}
+
+	// are we moving into an existing directory? Then we just need to prepend
+	// it's name to the source.
+	// a -> b
+	// - a/f1 -> b/a/f1
+	// a -> b
+	// - a -> b/a
+	if dstIsDir {
+		if !srcIsDir {
+			return path.Join(to, path.Base(src))
+		}
+		return path.Join(to, src)
+	}
+
+	// are we moving a simple file? that's easy
+	if !srcIsDir {
+		// otherwise we just rename a file to another name
+		return to
+	}
+
+	// move a/ b, where a is a directory with a trailing slash and b
+	// does not exist, i.e. move a to b
+	if strings.HasSuffix(from, "/") {
+		return path.Join(to, strings.TrimPrefix(src, from))
```

Comment:
I don't see any difference with the "else" path that returns the same thing, no?

Is this if clause necessary?